internal/services/mssql/mssql_database_resource.go

```diff
@@ -552,6 +552,8 @@ func resourceMsSqlDatabaseRead(d *pluginsdk.ResourceData, meta interface{}) erro
 				return err
 			}
 			d.Set("maintenance_configuration_name", maintenanceConfigId.ResourceName)
+		} else {
+			d.Set("maintenance_configuration_name", "SQL_Default")
```

Comment:
@sinbai this field can become un-computed and then we can conditionally set this to an empty string instead - since this can only be set when Elastic Pool's are involved?

Reply:
@tombuildsstuff  thanks for your feedback. For property `maintenance_configuration_name`, the behavior of the API as following:

-  When `maintenance_configuration_name` and `elastic_pool_id` are not specifiedAPI, the API returns the default value  `SQL_Default`  for `maintenance_configuration_name`  .
-   The `maintenance_configuration_name` could not be specified with a differrent value than the elastic pool's  `maintenance_configuration_name` .
-   When `elastic_pool_id` is specified , the API returns the default value depends on the value of elastic pool's  `maintenance_configuration_name`.  

Based on the above information, I assume that the computed could not be removed. Also I updated code to fix both acctest failure and [GH issue #19080](https://github.com/hashicorp/terraform-provider-azurerm/issues/19080), could you please take another look? Feel free to let me know your suggestions.

Test Results:
![image](https://user-images.githubusercontent.com/39109137/199437848-02ccfeb9-2fb1-4ece-9777-c1a800b42234.png)